frontend/src/workStation/Terminal.jsx

```diff
@@ -0,0 +1,76 @@
+import { Box, Text, Button, useToast } from "@chakra-ui/react";
+import axios from "axios";
+import { useState } from "react";
+import { useParams } from "react-router-dom";
+
+function Terminal({ editorRef, language, version, fileContent }) {
+  const [codeOutput, setCodeOutput] = useState(null);
+  const [isLoading, setisLoading] = useState(false);
+  const [isCodeBugged, setIsCodeBugged] = useState(false);
+
+  const BACKEND_URL = import.meta.env.VITE_BACKEND_URL;
+  const toast = useToast();
+  const { groupId, fileId } = useParams();
+
+  const handleRunCode = async () => {
+    setCodeOutput("");
+    const sourceCode = editorRef.current.getValue();
+    const params = {
+      language: language,
+      version: version,
+      codeBase: fileContent,
+    };
+    if (!sourceCode) return;
```

Comment:
Why do you return if this value is null when you don't use it later?

Reply:
The idea is to stop the function from going any further. I will throw an error instead.

Reply:
Why do you want to stop here though when `sourceCode` isn't sent to the backend to run the code? It looks like `fileContent` is the field that's required to not be null.

Reply:
@azhiah sourceCode is the current text within the editor at a given time. If I remove that line, you could clear the editor and click run and it would still produce an output. It is not good UX. That was my idea behind it. I can come up with a better optimization if you want me to keep the logic.

Reply:
If the editor was cleared, wouldn't `fileContent` be null?

Reply:
Nevermind, I watched the video and understand it now